File: GFG/Chicks in a Zoo/Chicks in a Zoo.cpp
class solution{
  public:
long long int NoOfChicks(int n){
	    // Code here
	  long long int popn[36];
	  long long int brth[36];
	  
	  popn[0]=0;popn[1]=1;
	  brth[0]=0;brth[1]=1;
	  
	  for(int i=2;i<=6;i++){
	      brth[i]=2*(popn[i-1]-0);
	      popn[i]=brth[i]+popn[i-1];
	  }
	  
	  for(int i=7;i<=n;i++){
	      brth[i]=2*(popn[i-1]-brth[i-6]);
	      popn[i]=popn[i-1]+brth[i]-brth[i-6];
	  }
	  return popn[n];
	}
};

File: GFG/Chicks in a Zoo/Chicks in a Zoo.java
class Solution
{
    public long NoOfChicks(int N)
    {
        // Code here
        long[]dp=new long[N+1];//new born
        dp[1]=1;
        long total=1;
        for(int i=2;i<=N;i++){
            if(i>6){
                total-=dp[i-6];
            }
            dp[i]=total*2;
            total+=(total*2);
        }
        return total;
    }
}
